File: src/main/java/com/partnershipmanagement/Entities/Assessment.java
package com.partnershipmanagement.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Entity
@Getter
@Setter
@ToString
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Assessment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    int idAssessment;
    float score;
    String feedback;

    @Enumerated(EnumType.STRING)
    Status status;

    @Enumerated(EnumType.STRING)
    AcceptanceStatus acceptanceStatus;

    boolean adminAcceptance;
    boolean partnerAacceptance;

    @JsonIgnore
    @ManyToOne
    Partnership partnership;
}

File: FrontEnd/Eduport-Angular/src/app/PartnershipManagement/Models/Assessment.ts
import { Partnership } from "./Partnership";
import { Status } from "./Status";
import {AcceptanceStatus} from "./AcceptanceStatus"


export interface Assessment {
    idAssessment: number;
    score: number;
    feedback: string;
    status: Status;
    acceptanceStatus: AcceptanceStatus;
    adminAcceptance: boolean;
    partnerAcceptance: boolean;
    partnership: Partnership;
  }
